emulator.py

```diff
@@ -26,26 +39,116 @@ def __init__(self):
         self.y = []
         self.number_of_samples = [] # can add any other quantities of interest here
         self.val_loss = []
+        self.models = defaultdict(dict)
         
-    def read_data_from_folder(self):
-        # TODO: Use np.loadtxt so that we can append to self.x and self.y for training --> will depend on formatting of input
-        pass
-                
-    def train_random_forest_regressor(self, scale = False, n_estimators = 100):
+
+    def read_data(self, training_file, test_file, scale=False, normalize_y=False):
+        self.scale = scale
+        self.normalize_y = normalize_y
+
+        infile_train = open(training_file,'rb')
+        new_dict_train = pickle.load(infile_train)
+        infile_train.close()
+        
+        input_train = new_dict_train['input_data']
+        output_train = new_dict_train['output_data']
+
+        self.xs_train_orig = np.array(input_train.drop(columns=['object_id']))
+        self.ys_train_orig = np.array(output_train.drop(columns=['object_id']))
+        self.extra_train = new_dict_train['extra_input']
+        self.r_vals = self.extra_train['r_vals']
+
         if scale:
             self.scaler = StandardScaler()
-            self.scaler.fit(self.x)
-            self.x_scaled = self.scaler.transform(self.x)
-            X_train, X_test, y_train, y_test = train_test_split(self.x_scaled, self.y)
+            self.scaler.fit(self.xs_train_orig)
+            self.xs_train = self.scaler.transform(self.xs_train_orig)
         else:
-            X_train, X_test, y_train, y_test = train_test_split(self.x, self.y)
-        self.model_rf = RandomForestRegressor(verbose = 0, n_jobs=-1, n_estimators=n_estimators) #n_jobs=-1 parallelises the training
-        self.model_rf.fit(X_train, y_train)
-        y_pred = self.model_rf.predict(X_test)
-        val_loss = mean_squared_error(y_test, y_pred)
-        print("Validation Loss: " + str(val_loss))
-        self.val_loss_list.append(val_loss)
+            self.xs_train = self.xs_train_orig
+
+        if normalize_y:
+            self.y_mean = np.mean(self.ys_train_orig, axis=0)
+            self.ys_train = self.ys_train_orig/self.y_mean
+        else:
+            self.ys_train = self.ys_train_orig
+        
+        infile_test = open(test_file,'rb')
+        new_dict_test = pickle.load(infile_test)
+        infile_test.close()
         
+        input_test = new_dict_test['input_data']
+        output_test = new_dict_test['output_data']
+
+        self.xs_test_orig = np.array(input_test.drop(columns=['object_id']))
+        self.ys_test_orig = np.array(output_test.drop(columns=['object_id']))    
```

Comment:
I understand why you do this but I actually use only `input_test.drop(columns=['object_id'])`, which allows the columns name to stay there.

I suggest keeping this on but if we have time, can I go through this and see if your emulator can also use as input the same as mine?